contracts/Staking.sol

```diff
@@ -344,27 +367,33 @@ contract Staking is Ownable, ReentrancyGuard {
     function _withdrawProviderTokens(
         uint256 _amount,
         address _tokenAddress,
-        uint256 _nextVerifiablePeriod
+        uint256 _nextVerifiablePeriod,
+        bool _contractFinished
     ) internal onlyAllowedToken(_tokenAddress) nonReentrant {
-        require(
-            lastProviderStake[msg.sender] < _nextVerifiablePeriod,
-            'Provider lock-up until the next verification.'
-        );
+        if (!_contractFinished) {
+            require(
+                lastProviderStake[msg.sender] < _nextVerifiablePeriod,
+                'Provider lock-up until the next verification.'
+            );
 
-        require(
-            providerPool[_tokenAddress].sub(_amount) >=
-                usersPool[_tokenAddress].mul(leverage),
-            'Withdrawal exceeds leveraged cap.'
-        );
+            // Allow provider withdrawal as long as the provider user pool exceeds the provider pool
```

Comment:
Small wording mistake should be `Allow provider withdrawal as long as the provider pool exceeds the user pool`

Reply:
Fixed!